### PR TITLE
MINIFI-222: Upgrade spdlog to latest release.

### PR DESCRIPTION
**Pull request by phrocker**
Modified Logger so that it removed the previously required flush argument.

Copyright headers changed but copyright license is the same.

### CODE REVIEW

**Comment by apiri** [comment]
hey @phrocker,

Just so I can better scope out licensing and such, is this version 0.12?

**Comment by apiri** [comment]
I see that it is in the code and is 0.12 (https://github.com/apache/nifi-minifi-cpp/pull/55/files#diff-ec88af83a0f1976c3c605e8b1027c454R10)

File: include/spdlog/sinks/ansicolor_sink.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
+//
+// Copyright(c) 2016 Kevin M. Godby (a modified version by spdlog).
```
Comment (apiri) [inline include/spdlog/sinks/ansicolor_sink.h]: Seems like we would need to call this out similar to https://github.com/apache/nifi-minifi-cpp/blob/ff7aa7fc3bf55681398427f1de6af7f352a100fa/LICENSE#L394

File: include/spdlog/sinks/dist_sink.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
+//
+// Copyright (c) 2015 David Schury, Gabi Melman
```
Comment (apiri) [inline include/spdlog/sinks/dist_sink.h]: Believe we need to include David Schury in the LICENSE in addition to Gabi Melman: https://github.com/apache/nifi-minifi-cpp/blob/ff7aa7fc3bf55681398427f1de6af7f352a100fa/LICENSE#L243

File: include/spdlog/sinks/msvc_sink.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
+//
+// Copyright(c) 2016 Alexander Dalshov.
```
Comment (apiri) [inline include/spdlog/sinks/msvc_sink.h]: Also should be covered in LICENSE

File: include/spdlog/sinks/wincolor_sink.h [hunk 4]
```diff
@@ -0,0 +1,116 @@
+//
+// Copyright(c) 2016 spdlog
```
Comment (apiri) [inline include/spdlog/sinks/wincolor_sink.h]: LICENSE adjustment for this entity as well

Comment (phrocker) [inline include/spdlog/sinks/wincolor_sink.h]: thanks!

**Comment by phrocker** [comment]
I put the authors with the corresponding license multiple times. I feel like it would be better to include the Copyrights and the license once, but I've seen it done like this in other apache projects. advise if the alternative is preferred.

**Comment by apiri** [comment]
I think the license once is preferred to kind of limit the noise and still provides the full coverage.  Reviewing the ListenHTTP I saw CivetWeb (https://github.com/civetweb/civetweb/blob/master/LICENSE.md#included-with-all-features) seemed to do this and I think is clear from a legality standpoint about spdlog as a whole.

**Comment by apiri** [comment]
@phrocker changes look good, thanks, will get this merged

**Pull request closed without merging**